packages/hash/src/index.ts

```diff
@@ -2,62 +2,69 @@
 // Inspired by https://github.com/garycourt/murmurhash-js
 // Ported from https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash2.cpp#L37-L86
 
+const encoder = new TextEncoder()
+
+// bufferLength must be a multiple of 4 to satisfy Int32Array constraints
+let bufferLength = 2 * 1024
+let buffer = new ArrayBuffer(bufferLength)
+let uint8View = new Uint8Array(buffer)
+let int32View = new Int32Array(buffer)
```

Comment:
Reading bytes with one `.charCodeAt(i)` call per byte (plus the bitwise operations to shift it in place) is less efficient than just translate everything into a byte array directly with `TextEncoder`. Very small strings might perform worse, but for the production dataset I tested with this is a positive change.

Reply:
Also `.charCodeAt()` isn't technically correct as it reads utf-16 code units, not bytes.